src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigDocumentOptionsProviderFactory.cs

```diff
@@ -6,32 +6,24 @@
 
 using System;
 using System.Collections.Immutable;
-using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.ErrorLogger;
 
 namespace Microsoft.CodeAnalysis.Options.EditorConfig
 {
-    [Export(typeof(IDocumentOptionsProviderFactory)), Shared]
-    [ExportMetadata("Name", PredefinedDocumentOptionsProviderNames.EditorConfig)]
-    internal sealed class EditorConfigDocumentOptionsProviderFactory : IDocumentOptionsProviderFactory
+    internal static class EditorConfigDocumentOptionsProviderFactory
     {
-        [ImportingConstructor]
-        public EditorConfigDocumentOptionsProviderFactory()
-        {
-        }
-
-        public IDocumentOptionsProvider? TryCreate(Workspace workspace)
+        public static IDocumentOptionsProvider? TryCreate(Workspace workspace)
         {
             if (!ShouldUseNativeEditorConfigSupport(workspace))
```

Comment:
@JoeRobich I was able to verify the integration test failures from this PR locally, and the root cause seems to be this code path (workspace constructor) can be invoked from a background thread, but fetching options can force loading of option persisters, which seem to require UI thread on creation. I moved this specific check `ShouldUseNativeEditorConfigSupport` from here down into `GetOptionsForDocumentAsync` below and then the tests pass locally. I think the correct long term solution here would be to remove the UI thread requirement from option persisters during their construction OR add functionality in GlobalOptionsService to be able to force UI thread before loading persisters. For now, this change should unblock us.

Reply:
By the way, do we still need to support legacy editorconfig provider? I would be fine with us deleting it completely as well. Tagging @jasonmalinowski